src/gpuarray_buffer_cuda.c

```diff
@@ -290,7 +270,48 @@ static void *cuda_init(int ord, int flags, int *ret) {
 
     return res;
 }
+static void *cuda_init(int ord, int flags, int *ret) {
+    CUdevice dev;
+    cuda_context *res;
+    static int init_done = 0;
 
+    if (ord == -2) {
+      CUcontext ctx;
+      /* Grab the ambient context */
+      err = cuCtxGetCurrent(&ctx);
+      CHKFAIL(NULL);
+      res = cuda_make_ctx(ctx, DONTFREE);
+      if (res == NULL) {
+        FAIL(NULL, GA_IMPL_ERROR);
+      }
+      res->flags |= flags;
+      return res;
```

Comment:
I'm not sure if we should or not.  If we already have a context then the driver API is probably initialized, though.